test/battle/ability/leaf_guard.c

```diff
@@ -50,29 +84,82 @@ SINGLE_BATTLE_TEST("Leaf Guard prevents status conditions from Flame Orb and Tox
             NONE_OF { MESSAGE("Leafeon was burned!"); STATUS_ICON(player, burn: TRUE); }
         }
         else {
-            NONE_OF { MESSAGE("Leafeon was badly poisoned!"); STATUS_ICON(player, poison: TRUE); }
+            NONE_OF { MESSAGE("Leafeon was badly poisoned!"); STATUS_ICON(player, badPoison: TRUE); }
         }
     }
 }
 
-TO_DO_BATTLE_TEST("Leaf Guard doesn't prevent status conditions from Flame Orb and Toxic Orb if Cloud Nine/Air Lock is on the field");
+SINGLE_BATTLE_TEST("Leaf Guard doesn't prevent status conditions from Flame Orb and Toxic Orb if Cloud Nine/Air Lock is on the field")
+{
+    u32 item, species, ability;
+    PARAMETRIZE { item = ITEM_FLAME_ORB; species = SPECIES_GOLDUCK;  ability = ABILITY_CLOUD_NINE; }
+    PARAMETRIZE { item = ITEM_TOXIC_ORB; species = SPECIES_GOLDUCK;  ability = ABILITY_CLOUD_NINE; }
+    PARAMETRIZE { item = ITEM_FLAME_ORB; species = SPECIES_RAYQUAZA; ability = ABILITY_AIR_LOCK; }
+    PARAMETRIZE { item = ITEM_TOXIC_ORB; species = SPECIES_RAYQUAZA; ability = ABILITY_AIR_LOCK; }
+    GIVEN {
+        ASSUME(gItemsInfo[ITEM_FLAME_ORB].holdEffect == HOLD_EFFECT_FLAME_ORB);
+        ASSUME(gItemsInfo[ITEM_TOXIC_ORB].holdEffect == HOLD_EFFECT_TOXIC_ORB);
+        PLAYER(SPECIES_LEAFEON) { Ability(ABILITY_LEAF_GUARD); Item(item); }
+        OPPONENT(species) { Ability(ability); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_SUNNY_DAY); }
+    } SCENE {
+        if (item == ITEM_FLAME_ORB) {
+            MESSAGE("Leafeon was burned!");
+            STATUS_ICON(player, burn: TRUE);
+        }
+        else {
+            MESSAGE("Leafeon was badly poisoned!");
+            STATUS_ICON(player, badPoison: TRUE);
+        }
+    }
+}
 
-SINGLE_BATTLE_TEST("Leaf Guard prevents Rest during sun")
+SINGLE_BATTLE_TEST("Leaf Guard prevents Rest during sun (Gen 5+)")
 {
+    u32 gen;
+    PARAMETRIZE { gen = GEN_4; }
+    PARAMETRIZE { gen = GEN_5; }
     GIVEN {
-        ASSUME(B_LEAF_GUARD_PREVENTS_REST >= GEN_5);
+        WITH_CONFIG(GEN_CONFIG_LEAF_GUARD_PREVENTS_REST, gen);
         ASSUME(GetMoveEffect(MOVE_REST) == EFFECT_REST);
         PLAYER(SPECIES_LEAFEON) { Ability(ABILITY_LEAF_GUARD); HP(100); MaxHP(200); }
         OPPONENT(SPECIES_WOBBUFFET);
     } WHEN {
         TURN { MOVE(opponent, MOVE_SUNNY_DAY); MOVE(player, MOVE_REST); }
     } SCENE {
-        MESSAGE("But it failed!");
-        NONE_OF {
+        if (gen >= GEN_5) {
+            MESSAGE("But it failed!");
```

Comment:
```suggestion
            NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_REST, player);
```
Prefer animation checks over text checks if possible.